### PR TITLE
Add/tagfragmentandactivity

### PR DESCRIPTION
**Pull request by haleyharelson**

### CODE REVIEW

File: app/src/main/java/com/fbu/fbuteam/activities/HomeActivity.java [hunk 1]
```diff
@@ -14,5 +19,27 @@ protected void onCreate(Bundle savedInstanceState) {
...
         setSupportActionBar(toolbar);
+
+        setLogoutBtnListener(toolbar);
+    }
+
+    private void setLogoutBtnListener(Toolbar toolbar) {
+        toolbar.setOnMenuItemClickListener(item -> {
+            switch (item.getItemId()) {
+                case R.id.miLogout:
+                    ParseUser.logOut();
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/activities/HomeActivity.java:30]: You can extract lines 30 to 33 in a separate function

File: app/src/main/java/com/fbu/fbuteam/activities/SignUpActivity.java [hunk 2]
```diff
@@ -56,9 +55,11 @@ private void signUpInBackground(ParseUser user) {
         user.signUpInBackground((error) -> {
             if (error == null) {
                 Toast.makeText(SignUpActivity.this, "Successfully signed up", Toast.LENGTH_SHORT).show();
-                goToHome();
+                goToTags();
             } else {
                 Log.e("SignUpActivity.", "Failed to sign up", error);
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/activities/SignUpActivity.java:61]: We can also extract lines 61 to 62 to a function called`handleSignupError`

File: app/src/main/java/com/fbu/fbuteam/activities/TagActivity.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+        if (currentUserSelection >= selectedBigIdeas.size()) {
+            Toast.makeText(getApplicationContext(), "Home Screen", Toast.LENGTH_SHORT).show();
+            goToHome();
+            return;
+        }
+
+        Node bigIdea = selectedBigIdeas.get(currentUserSelection);
+
+        DetailTagsFragment detailTagsFragment = DetailTagsFragment.newInstance(bigIdea);
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/activities/TagActivity.java:77]: Lines 74 to 78 could go to a function called "replaceFragment" or something like that

File: app/src/main/java/com/fbu/fbuteam/fragments/BigIdeasAdapter.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+    @NonNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(context).inflate(R.layout.bigidea_item, parent, false);
+        return new BigIdeasAdapter.ViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/fragments/BigIdeasAdapter.java]: We can make this function smaller. We can maybe create to functions: one that would contain lines 42 to 51, and the other would have lines 53 to 57

File: app/src/main/java/com/fbu/fbuteam/fragments/BigIdeasFragment.java [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+public class BigIdeasFragment extends Fragment {
+
+    private TextView textView1;
+    private TextView textView2;
+    private Button nextButton;
+    private BigIdeasAdapter adapter;
+    private LinearLayoutManager layoutManager;
+    private RecyclerView rvBigIdeas;
+    private ArrayList<Boolean> allTags = new ArrayList<>();
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/fragments/BigIdeasFragment.java]: Change static type to be List<Boolean> instead

File: app/src/main/java/com/fbu/fbuteam/fragments/BigIdeasFragment.java [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+    private ArrayList<Boolean> selectedTags = new ArrayList<>();
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/fragments/BigIdeasFragment.java]: Change static type to be List<Boolean> instead

File: app/src/main/java/com/fbu/fbuteam/fragments/BigIdeasFragment.java [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+        View rootView =  inflater.inflate(R.layout.bigideas_fragment, container, false);
+        initializeObjects(rootView);
+        return rootView;
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        rvBigIdeas = (RecyclerView) view.findViewById(R.id.rvBigIdeas);
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/fragments/BigIdeasFragment.java:64]: I would extract lines 62 to 66 to another function, maybe `setBigIdeasAdapters`?

File: app/src/main/java/com/fbu/fbuteam/fragments/BigIdeasFragment.java [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+        this.callback = callback;
+    }
+
+    public interface OnNextClickListener {
+        void goToDetailTagsFragment(List<Node> selectedBigIdeas);
+    }
+
+    private void goToDetailTags() {
+        nextButton.setOnClickListener(view1 -> {
+            if (atLeastOneChecked()) {
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/fragments/BigIdeasFragment.java:99]: if `atLeastOneChecked && callback != null`, and then we can remove line 101

File: app/src/main/java/com/fbu/fbuteam/fragments/BigIdeasFragment.java [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+                    callback.goToDetailTagsFragment(selectedBigIdeas);
+                }
+            } else {
+                Toast.makeText(getContext(), "Please select at least one topic.", Toast.LENGTH_LONG).show();
+            }
+        });
+    }
+
+    private List<Node> getSelectedBigIdeas() {
+        List<Node> b = new ArrayList<>();
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/fragments/BigIdeasFragment.java]: `b` is not a good variable name

File: app/src/main/java/com/fbu/fbuteam/fragments/BigIdeasFragment.java [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+        return b;
+    }
+
+    private void queryNodes() {
+        Node.Query query = new Node.Query();
+        query.withRelations();
+        query.fromType("big-idea");
+        query.findInBackground((response, e) -> {
+            if (e == null) {
+                for (Node a : response) {
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/fragments/BigIdeasFragment.java]: `a` is not a good variable name

File: app/src/main/java/com/fbu/fbuteam/fragments/DetailTagsAdapter.java [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+        View view = LayoutInflater.from(context).inflate(R.layout.item_tag, parent, false);
+        return new ViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
+        Node bigIdea = selectedBigIdeas.get(position);
+        holder.tagBox.setText(bigIdea.getName());
+
+        holder.tagBox.setOnCheckedChangeListener((compoundButton, isChecked) -> {
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/fragments/DetailTagsAdapter.java]: I would put lines 43 to 52 in a separate function

File: app/src/main/java/com/fbu/fbuteam/fragments/DetailTagsAdapter.java [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+            if (listOfChecked.get(position)) {
+                holder.cardView.setCardBackgroundColor(Color.rgb(83, 29, 85));
+                holder.tagBox.setTextColor(Color.WHITE);
+            } else {
+                holder.cardView.setCardBackgroundColor(Color.WHITE);
+                holder.tagBox.setTextColor(Color.rgb(56, 14, 67));
+            }
+        });
+
+        for (int i = 0; i < bigIdea.getChildren().size(); i++) {
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/fragments/DetailTagsAdapter.java]: would also put this line in a separate function

File: app/src/main/java/com/fbu/fbuteam/fragments/DetailTagsFragment.java [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+
+    public static Button finishButton;
+    private TextView textView;
+    private OnNextClickListener callback;
+    private DetailTagsAdapter adapter;
+    private LinearLayoutManager layoutManager;
+    private RecyclerView rvTags;
+    private Node bigIdea;
+    private List<Node> bigIdeasList;
+    private ArrayList<Boolean> allTags = new ArrayList<>();
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/fragments/DetailTagsFragment.java]: change static type to List<>

File: app/src/main/java/com/fbu/fbuteam/fragments/DetailTagsFragment.java [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+    private ArrayList<Boolean> selectedTags = new ArrayList<>();
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/fragments/DetailTagsFragment.java]: change static type to List<>

File: app/src/main/java/com/fbu/fbuteam/fragments/DetailTagsFragment.java [hunk 15]
```diff
@@ -0,0 +1,144 @@
...
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.detailtags_fragment, container, false);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/fragments/DetailTagsFragment.java]: I would remove this empty line

File: app/src/main/java/com/fbu/fbuteam/fragments/DetailTagsFragment.java [hunk 16]
```diff
@@ -0,0 +1,144 @@
...
+        super.onViewCreated(view, savedInstanceState);
+        createRV(view);
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/fragments/DetailTagsFragment.java]: I would rename "createRV" to "createRecyclerView"

**Review by LeoNero — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/fbu/fbuteam/fragments/BigIdeasFragment.java (hunk 8 above)
Comment (haleyharelson) [reply app/src/main/java/com/fbu/fbuteam/fragments/BigIdeasFragment.java:99]: I want the else statement to only result from the first condition being false though

File: app/src/main/java/com/fbu/fbuteam/fragments/BigIdeasAdapter.java [hunk 17]
```diff
@@ -0,0 +1,87 @@
+package com.fbu.fbuteam.fragments;
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/fragments/BigIdeasAdapter.java]: You should create an `adapters` folder and put the `BigIdeasAdapter` and the `DetailsTagAdapter` there

File: app/src/main/AndroidManifest.xml [hunk 18]
```diff
@@ -13,7 +13,9 @@
         android:supportsRtl="true"
         android:theme="@style/AppTheme.NoActionBar"
         android:usesCleartextTraffic="true">
-        <activity android:name=".activities.LoginActivity">
+
+        <activity android:name=".activities.LoginActivity"></activity>
+        <activity android:name=".activities.TagActivity">
```
Comment (LeoNero) [inline app/src/main/AndroidManifest.xml]: The start activity should be Login, not Tag

File: app/src/main/java/com/fbu/fbuteam/fragments/BigIdeasAdapter.java [hunk 19]
```diff
@@ -0,0 +1,87 @@
...
+        }
+    }
+
+    @Override
+    public int getItemCount() {
+        return bigIdeas.size();
+    }
+
+    public static class ViewHolder extends RecyclerView.ViewHolder {
+
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/fragments/BigIdeasAdapter.java]: Maybe remove this empty line?

File: app/src/main/java/com/fbu/fbuteam/fragments/BigIdeasAdapter.java [hunk 20]
```diff
@@ -0,0 +1,87 @@
...
+        CheckBox bigIdeaBox;
+        CardView cardView;
+
+        public ViewHolder(@NonNull View itemView) {
+            super(itemView);
+            bigIdeaBox = (CheckBox) itemView.findViewById(R.id.bigIdeaBox);
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/fragments/BigIdeasAdapter.java]: You can remove the (Checkbox) for this line and the following

File: app/src/main/java/com/fbu/fbuteam/fragments/BigIdeasFragment.java [hunk 21]
```diff
@@ -0,0 +1,143 @@
...
+        return selectedBigIdeas;
+    }
+
+    private void queryNodes() {
+        Node.Query query = new Node.Query();
+        query.withRelations();
+        query.fromType("big-idea");
+        query.findInBackground((response, e) -> {
+            if (e == null) {
+                for (Node node: response) {
```
Comment (LeoNero) [inline app/src/main/java/com/fbu/fbuteam/fragments/BigIdeasFragment.java:128]: `for (Node node : response)`

File: app/src/main/res/layout/activity_home.xml [hunk 22]
```diff
@@ -10,11 +10,12 @@
         android:id="@+id/toolbar"
         android:layout_width="match_parent"
         android:layout_height="?attr/actionBarSize"
-        android:background="?attr/colorPrimary"
+        android:background="#A911A5"
```
Comment (LeoNero) [inline app/src/main/res/layout/activity_home.xml:13]: You can add this color to the `colors.xml` file

File: app/src/main/res/layout/bigideas_fragment.xml [hunk 23]
```diff
@@ -0,0 +1,80 @@
...
+        style="@style/Widget.MaterialComponents.Button.TextButton"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="8dp"
+        android:layout_marginLeft="8dp"
+        android:layout_marginTop="32dp"
+        android:layout_marginEnd="8dp"
+        android:layout_marginRight="8dp"
+        android:fontFamily="sans-serif-light"
+        android:text="What are you into?"
```
Comment (LeoNero) [inline app/src/main/res/layout/bigideas_fragment.xml:35]: You can add this string to the `strings.xml` file

File: app/src/main/res/layout/bigideas_fragment.xml [hunk 24]
```diff
@@ -0,0 +1,80 @@
...
+        android:layout_height="wrap_content"
+        android:layout_marginStart="32dp"
+        android:layout_marginLeft="32dp"
+        android:layout_marginTop="8dp"
+        android:layout_marginEnd="32dp"
+        android:layout_marginRight="32dp"
+        android:gravity="center"
+        android:fontFamily="sans-serif-light"
+        android:textColor="@color/white"
+        android:text="Tell us what topics you'd be interested in seeing news stories about."
```
Comment (LeoNero) [inline app/src/main/res/layout/bigideas_fragment.xml:56]: Same here

**Review by LeoNero — CHANGES_REQUESTED** [review]

**Review by ghost — DISMISSED** [review]

**Review by LeoNero — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**